src/languageclient.rs

```diff
@@ -2074,6 +2074,50 @@ pub trait ILanguageClient: IVim {
         Ok(())
     }
 
+    fn window_handleProgress(&self, params: &Option<Params>) -> Result<()> {
+        info!("Begin {}", NOTIFICATION__WindowProgress);
+        let params: WindowProgressParams = serde_json::from_value(params.clone().to_value())?;
+
+        let busy = !params.done.unwrap_or(false);
+
+        let mut buf = "LS: ".to_owned();
+
+        if busy {
+            // For RLS this can be "Build" or "Diagnostics" or "Indexing".
+            buf.push_str(
+                params
+                    .title
+                    .as_ref()
```

Comment:
You could use `.cloned()` to simply code here.

Reply:
This is necessary to turn `Option<String>` into an `Option<&str>` so I can `.unwrap_or(&'static str)` and prevent an extra allocation. An alternative would be `params.title.unwrap_or("Busy".into())` which would allocate for `"Busy"`.

Reply:
I see.